src/Model/Mail.php

```diff
@@ -277,8 +276,8 @@ public static function sendWall(array $recipient = [], $body = '', $subject = ''
 
 		$recip_handle = $recipient['nickname'] . '@' . substr(DI::baseUrl(), strpos(DI::baseUrl(), '://') + 3);
 
-		$sender_nick = basename($replyto);
-		$sender_host = substr($replyto, strpos($replyto, '://') + 3);
+		$sender_nick = basename($me['url']);
+		$sender_host = substr($me['url'], strpos($me['url'], '://') + 3);
 		$sender_host = substr($sender_host, 0, strpos($sender_host, '/'));
```

Comment:
These are the parts of the code that needs to be rewritten. This is code from a time where we hadn't got the `addr` value.

Reply:
So, just `$me['addr']` ?

Reply:
After review of the code, that's what I went with.

Reply:
That whole part is ugly and a candidate for `Contact::getByURL` in the next release.